mmv1/products/iamworkforcepool/api.yaml

```
@@ -126,3 +126,195 @@ objects:
+            * `attribute.{custom_attribute}`:
+              `principalSet://iam.googleapis.com/locations/{location}/workforcePools/{pool}/attribute.{custom_attribute}/{value}`
+
+          Each value must be a [Common Expression Language](https://opensource.google/projects/cel)
+          function that maps an identity provider credential to the normalized attribute specified
+          by the corresponding map key.
+
+          You can use the `assertion` keyword in the expression to access a JSON representation of
+          the authentication credential issued by the provider.
+
+          The maximum length of an attribute mapping expression is 2048 characters. When evaluated,
+          the total size of all mapped attributes must not exceed 8KB.
+
+          For OIDC providers, you must supply a custom mapping that includes the `google.subject` attribute.
+          For example, the following maps the sub claim of the incoming credential to the `subject` attribute
+          on a Google token:
+          ```
+          {"google.subject": "assertion.sub"}
+          ```
+
+          An object containing a list of `"key": value` pairs.
+          Example: `{ "name": "wrench", "mass": "1.3kg", "count": "3" }`.
+      - !ruby/object:Api::Type::String
+        name: 'attributeCondition'
+        description: |
+          A [Common Expression Language](https://opensource.google/projects/cel) expression, in
+          plain text, to restrict what otherwise valid authentication credentials issued by the
+          provider should not be accepted.
+          
+          The expression must output a boolean representing whether to allow the federation.
+          
+          The following keywords may be referenced in the expressions:
+            * `assertion`: JSON representing the authentication credential issued by the provider.
+            * `google`: The Google attributes mapped from the assertion in the `attribute_mappings`.
+              `google.profile_photo` and `google.display_name` are not supported.
+            * `attribute`: The custom attributes mapped from the assertion in the `attribute_mappings`.
+          
+          The maximum length of the attribute condition expression is 4096 characters.
+          If unspecified, all valid authentication credentials will be accepted.
+          
+          The following example shows how to only allow credentials with a mapped `google.groups` value of `admins`:
+          ```
+          "'admins' in google.groups"
+          ```
+      - !ruby/object:Api::Type::NestedObject
+        name: 'saml'
+        description: Represents a SAML identity provider.
+        exactly_one_of:
+          - saml
+          - oidc
+        properties:
+          - !ruby/object:Api::Type::String
+            name: idpMetadataXml
+            description: |
+              Required. SAML Identity provider configuration metadata xml doc.
```

Marking fields as optional / required will happen automatically.

```suggestion
             SAML Identity provider configuration metadata xml doc.
```

---

fixed